vcd/resource_vcd_independent_disk.go

```diff
@@ -141,16 +223,67 @@ func resourceVcdIndependentDiskRead(d *schema.ResourceData, meta interface{}) er
 		return fmt.Errorf(errorRetrievingOrgAndVdc, err)
 	}
 
-	_, err = vdc.QueryDisk(d.Get("name").(string))
+	identifier := d.Id()
+	var disk *govcd.Disk
+	if identifier != "" {
+		disk, err = vdc.GetDiskById(identifier, true)
+		if govcd.IsNotFound(err) {
+			log.Printf("unable to find disk with ID %s: %s. Removing from state", identifier, err)
+			d.SetId("")
+			return nil
+		}
+		if err != nil {
+			return fmt.Errorf("unable to find disk with ID %s: %s", identifier, err)
+		}
+	} else {
+		identifier = d.Get("name").(string)
+		disks, err := vdc.GetDisksByName(identifier, true)
+		if govcd.IsNotFound(err) {
+			log.Printf("unable to find disk with ID %s: %s. Removing from state", identifier, err)
```

Comment:
Search by name. The error should use the name

Reply:
updated